server/handlers/component_handler.go

```diff
@@ -79,49 +80,94 @@ func (h *Handler) GetAllComponents(rw http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// swagger:route GET /api/meshmodel/components MeshmodelGet idMeshmodelGet
-// Handle GET request for getting all meshmodel components.
-// Components can be further filtered through query parameter ?version=
+/**Meshmodel endpoints **/
+const DefaultPageSizeForMeshModelComponents = 25
+
+// swagger:route GET /api/meshmodel/model GetMeshmodelModels idGetMeshmodelModels
+// Handle GET request for getting all meshmodel models. The component type/model name should be lowercase like "kubernetes", "istio"
+// ?version={version} If version is unspecified then all models are returned
+// ?order={field} orders on the passed field
+// ?category={category of model} If category in unspecified then all models are returned
+// ?sort={[asc/desc]} Default behavior is asc
+// ?search={[true/false]} If search is true then a greedy search is performed
+// ?page={page-number} Default page number is 1
+// ?pagesize={pagesize} Default pagesize is 25
```

Comment:
@Revolyssup is there a value (e.g. "all" or "0") that is supported for unlimited pagesize?

Reply:
In that case, let’s agree that “all” means return all results.

Reply:
okay

Reply:
Done